core/frontend/src/store/video.ts

```diff
@@ -169,6 +171,36 @@ class VideoStore extends VuexModule {
       }))
   }
 
+  @Action
+  async fetchSDP(stream: StreamStatus): Promise<void> {
+    const { video_source } = stream.video_and_stream
+    // eslint-disable-next-line no-extra-parens
+    const source = (<VideoSourceLocal>video_source).Local.device_path || (<VideoSourceGst>video_source).Gst.source.Fake
+
+    back_axios({
+      method: 'get',
+      url: `${this.API_URL}/sdp?source=${encodeURIComponent(source)}`,
+      timeout: 10000,
+      responseType: 'text',
+    })
+      .then((response) => {
+        if (response.status === 200) {
+          const blob = new Blob([response.data], { type: 'application/sdp' })
+          const url = URL.createObjectURL(blob)
+          const a = document.createElement('a')
+          a.href = url
+          a.download = `${stream.video_and_stream.name}.sdp`
+          document.body.appendChild(a)
+          a.click()
+          document.body.removeChild(a)
+          URL.revokeObjectURL(url)
```

Comment:
You should use the [file-saver](https://www.npmjs.com/package/file-saver) library that's already available on the project